cmd/agent/app/reporter/grpc/collector_proxy.go

```diff
@@ -45,7 +45,7 @@ func NewCollectorProxy(builder *ConnBuilder, agentTags map[string]string, mFacto
 	grpcMetrics := mFactory.Namespace(metrics.NSOptions{Name: "", Tags: map[string]string{"protocol": "grpc"}})
 	return &ProxyBuilder{
 		conn:     conn,
-		reporter: reporter.WrapWithMetrics(NewReporter(conn, agentTags, logger), grpcMetrics),
+		reporter: reporter.WrapWithQueue(reporter.WrapWithMetrics(NewReporter(conn, agentTags, logger), grpcMetrics), logger),
```

Comment:
We need to rethink how metrics would work, I think this change breaks them. Every time wrapped reporter fails to submit, we'd increment the failed counter, even though the batch may be retried. And vice versa, once we give up retrying, there's no metric at all. I think we need to split failure metrics into (a) how many retries we're going (a sustained elevated level may be cause for alarm), and (b) how many spans/batches we're dropping even after retries.

Reply:
I can merge the metrics decorator to this one, that would at least allow more accurate reporting.